metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1837,8 +1872,13 @@ void generateLeaderAndIsrUpdates(String context,
         // from the target ISR, but we need to exclude it here too, to handle the case
         // where there is an unclean leader election which chooses a leader from outside
         // the ISR.
+        //
+        // If the caller passed a valid broker ID for brokerWithUncleanShutdown, rather than
+        // passing NO_LEADER, this node should not be an acceptable leader. We also exclude
+        // brokerWithUncleanShutdown from ELR and ISR.
         IntPredicate isAcceptableLeader =
-            r -> (r != brokerToRemove) && (r == brokerToAdd || clusterControl.isActive(r));
+            r -> (r != brokerToRemove && r != brokerWithUncleanShutdown)
+                && (r == brokerToAdd || clusterControl.isActive(r));
```

Comment:
Since our guards around ELR (if it's enabled or not) are in PartitionChangeBuilder, we need to make sure this logic is correct when ELR is not enabled due to MV.

Reply:
I disabled the handleUncleanShutdown if ELR is not enabled.